compiler/passes/analyses/track_volume.py

```diff
@@ -26,190 +26,201 @@ def analyze(self, program: Program) -> dict: # The main function of the class
+        if type(instruction) == Dispose:
+            self.handle_dispose(instruction)
+            return
 
-    possible_volumes = []
+        if type(instruction) == Mix:
+            self.handle_mix(instruction)
+            return
 
-    for i in range(len(instructions.uses)):
-        possible_volumes.append(min(self.variable_volume[instructions.uses[i]]['volumes']))
+        if type(instruction) == Split:
+            self.handle_split(instruction)
+            return
+        if type(instruction) == Phi:
+            self.handle_phi(instruction)
 
-    entry = dict() # The dict that will hold our new entry in the variable_volume ds
-    volumes = list() # The list that will holds the volumes stored at each index
+    # We assume that the variable being created here has the minimum possible volume. This way, the compiler only proceeds when volume is guaranteed to be correct
+    def handle_phi(self, instructions: IR):
+        possible_volumes = []
 
-    volumes.append(min(possible_volumes)) # Add the dispense'd quantity toi the volumes list. In this case we assume that size is always 1, so we only do it a single time.
+        for i in range(len(instructions.uses)):
 
-    # Build the entry dict
-    entry['size'] = 1
-    entry['volumes'] = volumes
+            possible_volumes.append(deepcopy(min(self.variable_volume[instructions.uses[i]]['volumes'])))
 
-    # Add the entry to the volume tracker
-    self.variable_volume[instructions.defs['name']] = entry
+        entry = dict()  # The dict that will hold our new entry in the variable_volume ds
+        volumes = list()  # The list that will holds the volumes stored at each index
 
+        volumes.append(min(possible_volumes))  # Add the dispense'd quantity toi the volumes list. In this case we assume that size is always 1, so we only do it a single time.
 
-def handle_dispense(self, instructions : IR):
+        # Build the entry dict
+        entry['size'] = 1
+        entry['volumes'] = volumes
 
-    quantity = self._program.symbol_table.get_local(instructions.defs['name'], "main").value.volume['quantity']
+        # Add the entry to the volume tracker
+        self.variable_volume[instructions.defs['name']] = entry
 
-    # Initialize the structures we will use later on
-    entry = dict() # The dict that will hold our new entry in the variable_volume ds
-    volumes = list() # The list that will holds the volumes stored at each index
+    def handle_dispense(self, instructions: IR):
+        name_length = len(instructions.defs['name'])
+        name_no_num = instructions.defs['name'][0 : name_length -1]
+        name_num_only = instructions.defs['name'][-1]
 
-    volumes.append(quantity) # Add the dispense'd quantity toi the volumes list. In this case we assume that size is always 1, so we only do it a single time.
+        quantity = self._program.symbol_table.get_local(instructions.defs['name'], "main").volumes[name_no_num][int(name_num_only) - 1]

```

Comment:
The comment I left in `ir_visitor` refers to this hacky access.  Please update accordingly using `iid` or some other unique identifier.